pkg/issuer/acme/http/http.go

```diff
@@ -309,3 +323,66 @@ func testReachability(ctx context.Context, url *url.URL, key string) error {
 
 	return nil
 }
+
+func resolve(ctx context.Context, addr string, dnsServers []string) (string, error) {
+	log := logf.FromContext(ctx)
+	host, port, err := net.SplitHostPort(addr)
+	if err != nil {
+		return "", err
+	}
+	for _, server := range dnsServers {
+		ip, ok, err := lookupHost(ctx, host, server)
+		if err != nil {
+			log.V(logf.DebugLevel).Info("failed to lookup host on server", "error", err, "server", server)
+			continue
+		}
+		if !ok {
+			continue
+		}
+		return ip + ":" + port, nil
+	}
+	return "", fmt.Errorf("[%s] no such host: %s", strings.Join(dnsServers, ", "), host)
+}
+
+func lookupHost(ctx context.Context, host, server string) (string, bool, error) {
```

Comment:
I'm not sure this function is sufficient for HTTP-01 validation.

From my understanding, it will ask the resolver for all A, CNAME, and AAAA records.

It will look through every answer, and return the first A record it finds. If there are none, it will check all CNAME records until 1 results in an A or AAAA record and return that. If there are still no valid records, it will then return the first AAAA record it finds.

The issue is returning a single address rather than a slice of addresses to try. Let's encrypt will try multiple addresses if it cannot dial.

Reply:
I don't remember why I did go that way. I think I was having problems with `net.Resolver`. But I found an easier solution using it.